src/psyclone/domain/lfric/lfric_arg_descriptor.py

```diff
@@ -512,9 +513,9 @@ def _validate_scalar(self, arg_type):
 
         '''
         # Check whether something other than a scalar is passed in
-        if self._type not in LFRicArgDescriptor.VALID_SCALAR_NAMES:
+        if self._argument_type not in LFRicArgDescriptor.VALID_SCALAR_NAMES:
```

Comment:
Just a note to say this doesn't look right (although it's not the fault of this PR) - we shouldn't be comparing the 'type' of something with a list of 'valid scalar names'. I'd expect 'valid scalar types'. I'm not asking you to fix this - just making a note of it.

Reply:
I agree, it is clunky. As you said, perhaps as a part of another PR - it would be good to think about how the present `psyGen` classes/functions' names relate to names of similar structures in PSyIR.